manticore/core/cpu/abstractcpu.py

```diff
@@ -351,6 +371,32 @@ def syscall_number(self):
         '''
         raise NotImplementedError
 
+    def get_argument_values(self, model, prefix_args):
+        self._last_arguments = super(SyscallAbi, self).get_argument_values(model, prefix_args)
+        return self._last_arguments
+
+    def invoke(self, model, prefix_args=None):
+        max_arg_expansion = 32
+        # invoke() will call get_argument_values()
+        self._last_arguments = ()
+
+        ret = super(SyscallAbi, self).invoke(model, prefix_args)
+
+        if platform_logger.isEnabledFor(logging.DEBUG):
```

Comment:
this code will probably produce slightly incorrect output sometimes
- if a pointer doesn't point to a string. maybe we can check whether the bytes lie in ASCII
- if an argument is actually a negative signed int, it will print it out as unsigned

i think the benefits are worth those, likely infrequent, misprints, so we don't necessarily have to do anything.

maybe we can print the return code as both decimal and hex? for cases like:

```
sys_brk(0x6f01c0) -> 7274944
sys_brk(0x6f1000) -> 7278592

```

Reply:
So at one point, I considered producing `hexdump`-like output, but realized it might be adding complexity where it's not needed; I ended up adding a minimum length on strings in its place, with the assumption that it might catch some misprints.

Re: both decimal and hex output, one thing I considered is keeping it decimal if it's < 128, and hex otherwise. Or print it in decimal always, but add a parenthetical if it's big enough (`65280 (0xff00)`).